pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -318,6 +321,17 @@ private void updateSegmentMetrics(String tableNameWithType, TableConfig tableCon
         continue;
       }
 
+      long startTimeMs = segmentZKMetadata.getStartTimeMs();
+      boolean isStartTimeMsInvalid = startTimeMs > 0 && !TimeUtils.timeValueInValidRange(startTimeMs);
+      long endTimeMs = segmentZKMetadata.getEndTimeMs();
+      boolean isEndTimeMsInvalid = endTimeMs > 0 && !TimeUtils.timeValueInValidRange(endTimeMs);
+      if (isStartTimeMsInvalid) {
+        segmentsInvalidStartTime.add(segment);
+      }
+      if (isEndTimeMsInvalid) {
+        segmentsInvalidEndTime.add(segment);
+      }
```

Comment:
Let's only skip the `IN_PROGRESS` segments and count negative start/end time as invalid:
```
    ...
  }
```
```suggestion
      if (segmentZKMetadata.getStatus() != Status.IN_PROGRESS) {
        if (!TimeUtils.timeValueInValidRange(segmentZKMetadata.getStartTimeMs()) {
          segmentsInvalidStartTime.add(segment);
        }
        if (!TimeUtils.timeValueInValidRange(segmentZKMetadata.getEndTimeMs()) {
          segmentsInvalidEndTime.add(segment);
        }
      }
```